tests/utils/test_qtutils.py

```diff
@@ -58,69 +59,64 @@ class CheckOverflowTests(unittest.TestCase):
                   (float(INT64_MAX + 1), INT64_MAX)],
     }
 
-    def test_good_values(self):
+    @pytest.mark.parametrize('ctype, val', [(ctype, val) for ctype, vals in
+                                            GOOD_VALUES.items() for val in
+                                            vals])
+    def test_good_values(self, ctype, val):
         """Test values which are inside bounds."""
-        for ctype, vals in self.GOOD_VALUES.items():
-            for val in vals:
-                with self.subTest(ctype=ctype, val=val):
-                    qtutils.check_overflow(val, ctype)
+        qtutils.check_overflow(val, ctype)
 
-    def test_bad_values_fatal(self):
+    @pytest.mark.parametrize('ctype, val',
+                             [(ctype, val) for ctype, vals in
+                              BAD_VALUES.items() for (val, _) in vals])
+    def test_bad_values_fatal(self, ctype, val):
         """Test values which are outside bounds with fatal=True."""
-        for ctype, vals in self.BAD_VALUES.items():
-            for (val, _) in vals:
-                with self.subTest(ctype=ctype, val=val):
-                    with self.assertRaises(OverflowError):
-                        qtutils.check_overflow(val, ctype)
+        with pytest.raises(OverflowError):
+            qtutils.check_overflow(val, ctype)
 
-    def test_bad_values_nonfatal(self):
+    @pytest.mark.parametrize('ctype, val, repl',
+                             [(ctype, val, repl) for ctype, vals in
+                              BAD_VALUES.items() for (val, repl) in vals])
```

Comment:
I feel that this is close to being too complex. :worried: You don't want to *decrypt* test data...